### PR TITLE
1378593: concurrent content creates

### PR DESCRIPTION
**Pull request by vritant**

### CODE REVIEW

**Comment by barnabycourt** [comment]
@vritant Wouldn't this still have risk of collision when Candlepin is being run in cluster?

**Comment by vritant** [comment]
Good point, Ive updated the code. if this makes sense, I can fix master as well, as it was a backport.

**Comment by barnabycourt** [comment]
Only thing that needs to be double checked is that the system will work well with IDs that are uuid based as opposed to integer based (even if they are represented as strings). If everything checks out on this then please forward port to master as well.

File: server/spec/content_resource_spec.rb [hunk 1]
```diff
@@ -56,4 +56,19 @@
     # Verify the serial has changed (the entitlement has been regenerated)
     original_serial.should_not == new_serial
   end
+
+  it 'should handle concurrent content creates' do
+    threads = []
+    for i in 0..300
```
Comment (awood) [inline server/spec/content_resource_spec.rb]:
```ruby
300.times do
  threads << Thread.new do 
    @cp.create_content("cname", nil, random_string("clabel"), "ctype", "cvendor", {}, true)
  end
...
```

**Pull request merged**